src/commonMain/kotlin/com/charleskorn/kaml/YamlObjectInput.kt

```diff
@@ -36,11 +36,27 @@ internal class YamlObjectInput(map: YamlMap, yaml: Yaml, context: SerializersMod
 
             val currentEntry = entriesList[nextIndex]
             currentKey = currentEntry.key
-            val fieldDescriptorIndex = descriptor.getElementIndex(propertyName)
+
+            val pairedPropertyNames = (0 until descriptor.elementsCount)
+                .asSequence()
+                .map(descriptor::getElementName)
+                .map { elementName ->
+                    val yamlName = configuration.yamlNamingStrategy?.serialNameForYaml(elementName) ?: elementName
+                    elementName to yamlName
+                }
+
+            val fieldDescriptorIndex = pairedPropertyNames
```

Comment:
This might be simpler if `pairedPropertyNames` was a map of converted names to their field index